File: microbit-remote_ID2.js
let robot_id = 2
if (true) {
    radio.setGroup(robot_id)
}
let status = 0
basic.showIcon(IconNames.Happy)
basic.forever(function () {
    if (input.isGesture(Gesture.LogoUp)) {
        status = 1
        radio.sendValue("forward", 100)
    } else if (input.isGesture(Gesture.LogoDown)) {
        status = 2
        radio.sendValue("back", 100)
    } else if (input.isGesture(Gesture.TiltLeft)) {
        status = 3
        radio.sendValue("left", 100)
    } else if (input.isGesture(Gesture.TiltRight)) {
        status = 4
        radio.sendValue("right", 100)
    } else if (input.buttonIsPressed(Button.AB)) {
        status = 5
        radio.sendValue("dance1", 100)
    } else if (input.buttonIsPressed(Button.A)) {
        status = 6
        radio.sendValue("dance2", 100)
    } else if (input.buttonIsPressed(Button.B)) {
        status = 7
        radio.sendValue("dance3", 100)
    } else if (input.isGesture(Gesture.ScreenUp)) {
        status = 0
        radio.sendValue("stop", 100)
    }
})
basic.forever(function () {
    if (status == 1) {
        basic.showLeds(`
            . . # . .
            . # # # .
            # . # . #
            . . # . .
            . . # . .
            `)
    } else if (status == 2) {
        basic.showLeds(`
            . . # . .
            . . # . .
            # . # . #
            . # # # .
            . . # . .
            `)
    } else if (status == 3) {
        basic.showLeds(`
            . . # . .
            . # . . .
            # # # # #
            . # . . .
            . . # . .
            `)
    } else if (status == 4) {
        basic.showLeds(`
            . . # . .
            . . . # .
            # # # # #
            . . . # .
            . . # . .
            `)
    } else if (status == 5) {
        basic.showNumber(0)
    } else if (status == 6) {
        basic.showNumber(1)
    } else if (status == 7) {
        basic.showNumber(2)
    } else if (status == 0) {
        basic.showIcon(IconNames.Happy)
    }
})

File: microbit-remote_ID2.py
robot_id = 2
if True:
    radio.set_group(robot_id)
status = 0
basic.show_icon(IconNames.HAPPY)

def on_forever():
    global status
    if input.is_gesture(Gesture.LOGO_UP):
        status = 1
        radio.send_value("forward", 100)
    elif input.is_gesture(Gesture.LOGO_DOWN):
        status = 2
        radio.send_value("back", 100)
    elif input.is_gesture(Gesture.TILT_LEFT):
        status = 3
        radio.send_value("left", 100)
    elif input.is_gesture(Gesture.TILT_RIGHT):
        status = 4
        radio.send_value("right", 100)
    elif input.button_is_pressed(Button.AB):
        status = 5
        radio.send_value("dance1", 100)
    elif input.button_is_pressed(Button.A):
        status = 6
        radio.send_value("dance2", 100)
    elif input.button_is_pressed(Button.B):
        status = 7
        radio.send_value("dance3", 100)
    elif input.is_gesture(Gesture.SCREEN_UP):
        status = 0
        radio.send_value("stop", 100)
basic.forever(on_forever)

def on_forever2():
    if status == 1:
        basic.show_leds("""
            . . # . .
            . # # # .
            # . # . #
            . . # . .
            . . # . .
            """)
    elif status == 2:
        basic.show_leds("""
            . . # . .
            . . # . .
            # . # . #
            . # # # .
            . . # . .
            """)
    elif status == 3:
        basic.show_leds("""
            . . # . .
            . # . . .
            # # # # #
            . # . . .
            . . # . .
            """)
    elif status == 4:
        basic.show_leds("""
            . . # . .
            . . . # .
            # # # # #
            . . . # .
            . . # . .
            """)
    elif status == 5:
        basic.show_number(0)
    elif status == 6:
        basic.show_number(1)
    elif status == 7:
        basic.show_number(2)
    elif status == 0:
        basic.show_icon(IconNames.HAPPY)
basic.forever(on_forever2)
